From 26e6fbff9caff4f14dc41b3dac5b1c2ce08da026 Mon Sep 17 00:00:00 2001
From: Marco Bettini <marco.bettini@open-xchange.com>
Date: Tue, 16 Dec 2025 09:47:03 +0000
Subject: [PATCH 3/4] auth: Make the default
 passdb_static/passdb_default_password_scheme = PLAIN explicit

in preperation for the change of the global passdb_default_password_scheme default
---
 src/auth/auth-settings.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/auth/auth-settings.c b/src/auth/auth-settings.c
index fa94044df8..e6fb6a833d 100644
--- a/src/auth/auth-settings.c
+++ b/src/auth/auth-settings.c
@@ -268,11 +268,17 @@ static const struct auth_static_settings auth_static_default_settings = {
 	.userdb_static_allow_all_users = FALSE,
 };
 
+static const struct setting_keyvalue auth_static_default_settings_keyvalue[] = {
+       { "passdb_static/passdb_default_password_scheme", "PLAIN" },
+       { NULL, NULL }
+};
+
 const struct setting_parser_info auth_static_setting_parser_info = {
 	.name = "auth_static",
 
 	.defines = auth_static_setting_defines,
 	.defaults = &auth_static_default_settings,
+	.default_settings = auth_static_default_settings_keyvalue,
 
 	.struct_size = sizeof(struct auth_static_settings),
 	.pool_offset1 = 1 + offsetof(struct auth_static_settings, pool),
-- 
2.52.0

